azext_iot/tests/digitaltwins/test_dt_import_jobs_int.py

```diff
@@ -0,0 +1,177 @@
+            )
+        ).get_output_in_json()
+
+        assert_import_job_creation(create_valid_import_job_output)
+        valid_import_job_id = create_valid_import_job_output["id"]
+
+        # Show import job
+        show_import_job_output = self.cmd(
+            "dt job import show -n '{}' -g '{}' -j '{}'".format(instance_name, self.rg, valid_import_job_id)
+        ).get_output_in_json()
+        assert(show_import_job_output["id"] == valid_import_job_id)
+
+        # Create import job for invalid import data
+        invalid_import_job_output_filename = "{}_invalid_import_job_output.txt".format(instance_name)
+        create_invalid_import_job_output = self.cmd(
+            "dt job import create -n '{}' -g '{}' --df '{}' --ibc '{}' --isa '{}' --of '{}'".format(
+                instance_name, self.rg, invalid_import_data_file.rsplit('/', maxsplit=1)[-1],
+                self.storage_container, self.storage_account_name, invalid_import_job_output_filename
+            )
+        ).get_output_in_json()
+
+        assert_import_job_creation(create_invalid_import_job_output)
+        invalid_import_job_id = create_invalid_import_job_output["id"]
+
+        # List import jobs
+        list_import_jobs_output = self.cmd(
+            "dt job import list -n '{}' -g '{}'".format(instance_name, self.rg)
+        ).get_output_in_json()
+        assert len(list_import_jobs_output) == 2
+        import_job_ids = [valid_import_job_id, invalid_import_job_id]
+        assert list_import_jobs_output[0]["id"] in import_job_ids
+        assert list_import_jobs_output[1]["id"] in import_job_ids
+
+        # Poll to ensure desired status of import jobs
+        num_tries = 0
+        max_tries = 5
+        while num_tries < max_tries:
+            num_tries += 1
+            show_valid_import_job_output = self.cmd(
+                "dt job import show -n '{}' -g '{}' -j '{}'".format(instance_name, self.rg, valid_import_job_id)
+            ).get_output_in_json()
+            assert(show_valid_import_job_output["status"] != "failed")
+            show_invalid_import_job_output = self.cmd(
+                "dt job import show -n '{}' -g '{}' -j '{}'".format(instance_name, self.rg, invalid_import_job_id)
+            ).get_output_in_json()
+            assert(show_invalid_import_job_output["status"] != "succeeded")
+            if show_valid_import_job_output["status"] == "succeeded" and show_invalid_import_job_output["status"] == "failed":
+                assert(show_invalid_import_job_output["error"]["error"]["code"] == "DTDLParsingError")
+                assert(show_valid_import_job_output["error"] is None)
+                break
+            sleep(30)
+
+        list_models_output = self.cmd(
+            "dt model list -n {} -g {} --definition".format(instance_name, self.rg)
+        ).get_output_in_json()
+        assert len(list_models_output) == 10
+        for model in list_models_output:
+            assert model["id"]
```

Comment:
why not check that the right model and twin exist? Make a helper function to parse the ndjson and make a list of the models and twins so you can check

Reply:
I think writing an ndjson parser for the purposes of a test is a bit of over-engineering. The test can ensure that all the expected model ids twin ids and have been imported to the DT instance